### PR TITLE
Add new Smile.js version of points slider at checkout for Shopify

### PR DESCRIPTION
**Pull request by kingscott**

### CODE REVIEW

File: shopify-points-slider/smile-points-slider.liquid [hunk 1]
```diff
@@ -81,28 +81,31 @@
...
         return;
       }
 
-      var $sliderContainer = buildPointsSliderOrderSummarySection(SmileShopify.smile.customer);
+      var $sliderContainer = buildPointsSliderOrderSummarySection(smile.customer);
       var $pointsSlider = $sliderContainer.find('.sweettooth-points-slider');
       var pointsSliderData = $pointsSlider.data();
       var pointsProductId = pointsSliderData.pointsProductId;
 
-      SmileShopify.smile.fetchPointsProduct(SmileShopify.channel_key, pointsProductId).then(function(pointsProduct) {
```
Comment (kingscott) [inline shopify-points-slider/smile-points-slider.liquid]: Have you tested this yet? I believe you do [need](https://github.com/smile-io/smile-js/blob/master/src/methods/fetchPointsProduct.js#L5) a `channel_key`, unless the API doesn't need it.

File: shopify-points-slider/smile-points-slider.liquid [hunk 2]
```diff
@@ -212,20 +215,20 @@
           $redeemButton.addClass('btn--loading');
 
           // Hit the Smile.io API to spend customer's points
-          SmileShopify.smile.purchasePointsProduct(SmileShopify.channel_key, SmileShopify.smile.customer_auth_token, pointsProductId, { points_to_spend: sliderValue }).then(function(pointsPurchase) {
```
Comment (kingscott) [inline shopify-points-slider/smile-points-slider.liquid]: Similar with [this](https://github.com/smile-io/smile-js/blob/master/src/methods/purchasePointsProduct.js#L10) one.

File: shopify-points-slider/smile-points-slider.liquid [hunk 3]
```diff
@@ -212,20 +215,20 @@
...
 
             // Immediately update the customers points balance
-            $('.sweettooth-points-balance').text(SweetTooth.customer.points_balance - pointsPurchase.points_spent);
+            $('.sweettooth-points-balance').text(smile.customer.points_balance - pointsPurchase.points_spent);
 
             /**
              * Update customer's points balance based on the server's value.
              * This should be the same as the above but always nice to confirm.
              */
-            SmileShopify.smile.fetchCustomer(SmileShopify.channel_key, SmileShopify.smile.customer.id, SmileShopify.smile.customer_auth_token).then(function(customer) {
```
Comment (kingscott) [inline shopify-points-slider/smile-points-slider.liquid]: And finally just making a note of [this](https://github.com/smile-io/smile-js/blob/master/src/methods/fetchCustomer.js#L4) one.

File: shopify-points-slider/smile-points-slider.liquid (hunk 1 above)
Comment (ClaytonPassmore) [reply shopify-points-slider/smile-points-slider.liquid]: The method exposed by SmileShopify doesn't take a channel key. That's just the method used internally, and the channel key gets added by the SmileShopify instance.

**Comment by ClaytonPassmore** [comment]
@kingscott these method exposed by SmileShopify don't take a channel key. That's just for the methods used internally. The channel key gets added by the SmileShopify instance.

**Comment by kingscott** [comment]
@ClaytonPassmore Ahh, yeah. My bad. Was looking at the wrong repo 😭

**Comment by kingscott** [comment]
@ClaytonPassmore I can't approve my own PR - it looks good 🙌

**Pull request merged**